src/main/java/com/github/manolo8/darkbot/modules/utils/NpcAttacker.java

```diff
@@ -143,9 +145,17 @@ private boolean shouldSab() {
     }
 
     private boolean shouldRsb() {
-        if (!main.config.LOOT.RSB.ENABLED || !target.npcInfo.extra.has(NpcExtra.USE_RSB)) return false;
-        if (usedRsb < System.currentTimeMillis() - main.config.LOOT.RSB.AMMO_REFRESH) usedRsb = System.currentTimeMillis();
-
+        if (!main.config.LOOT.RSB.ENABLED || main.config.LOOT.RSB.KEY == null
+                || !target.npcInfo.extra.has(NpcExtra.USE_RSB)) return false;
+        String itemId = main.facadeManager.slotBars.standardBar.slots.stream()
+                .filter(slot -> Objects.equals(main.facadeManager.settings.getCharCode(valueOf
+                        ("SLOTBAR_" + (slot.slotNumber == 10 ? 0 : slot.slotNumber))), main.config.LOOT.RSB.KEY))
+                .findFirst().map(slot -> slot.item != null ? slot.item.id : null).orElse(null);
+        if (!Objects.equals(itemId, "ammunition_laser_rsb-75")
+                && !Objects.equals(itemId, "ammunition_laser_rcb-140")) return false; // TODO: Check bool cooldown from enum, after API gets merged.
```

Comment:
You wanted to restrict it because you told me you want it to not work if user sets a regular ammo like x1
And no, timer of item isn't distinguishable when it's not in cooldown. We can't automatically know if it's a cooldowned item or not.

Reply:
I never said i want it not to work. I said if something is wrong it should be apparent to the user. Making it directly never trigger is not apparent to the user at all to start with, it's simply disabling the feature

Reply:
With 1000ms It will be pretty annoying and cause spam around support channels. If it's RSB, it must be it. Would be okay if we could restrict it to items which have cooldowns.